### ISSUE TITLE
[Bug]: onDrop File.path = undefined in v32, works v31

### ISSUE THREAD

**Issue by Makio64** [electron/electron#43302]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a bug report that matches the one I want to file, without success.

### Electron Version

32.0.0.beta5

### What operating system(s) are you using?

macOS

### Operating System Version

MacOS Sorona 14.5

### What arch are you using?

x64

### Last Known Working Electron version

^31.x

### Expected Behavior

`console.log(file.path) // /Users/xxx/xxx.png`

### Actual Behavior

`console.log(file.path) // undefined`

**Comment by codebytere** [comment]
This was an intentional and communicated decision: https://github.com/electron/electron/issues/42053

**Comment by Makio64** [comment]
@codebytere I read the thread but I didnt see the replacement?

How do we get the path of the file now?

Electron is not web and knowing where the file dropped come from is crucial for example to save in the same folder. 

Also no deprecation cycle is harsh as its a breaking change for app using it.

**Comment by lutzroeder** [comment]
TL;DR use [getPathForFile](https://www.electronjs.org/docs/latest/api/web-utils) instead.

```javascript
electron.webUtils.getPathForFile(file)
```

**Comment by bbb169** [comment]
> electron
`import { webUtils } from 'electron';  `
hellow  while i am using this api to get file path, my renderer throw errors
`electron.js?v=71c22fc4:36 Uncaught TypeError: path.join is not a function
    at node_modules/electron/index.js (electron.js?v=71c22fc4:36:25)
    at __require2 (chunk-7D4SUZUM.js?v=71c22fc4:14:50)
    at electron.js?v=71c22fc4:54:16`


but while I change the modules import statement into require statement, this error is resolved, I think this api should support import statement
`const { webUtils } = require('electron');`

**Comment by bbb169** [comment]
> > electron
> > `import { webUtils } from 'electron';  `
> 
> hellow while i am using this api to get file path, my renderer throw errors
> 
> `electron.js?v=71c22fc4:36 Uncaught TypeError: path.join is not a function at node_modules/electron/index.js (electron.js?v=71c22fc4:36:25) at __require2 (chunk-7D4SUZUM.js?v=71c22fc4:14:50) at electron.js?v=71c22fc4:54:16`
> 
> but while I change the modules import statement into require statement, this error is resolved, I think this api should support import statement `const { webUtils } = require('electron');`

maybe you can use my repository to reproduce this， these codes is at `src\renderer\components\fileDragger\index.tsx`
my repository link is https://github.com/bbb169/subtitle-scraper-desktop/blob/48ab0be32bf47c0c994aa2d44c85afad21c917ba/src/renderer/components/fileDragger/index.tsx#L6  @lutzroeder

**Comment by darknblack** [comment]
> > electron
> > `import { webUtils } from 'electron';  `
> 
> hellow while i am using this api to get file path, my renderer throw errors
> 
> `electron.js?v=71c22fc4:36 Uncaught TypeError: path.join is not a function at node_modules/electron/index.js (electron.js?v=71c22fc4:36:25) at __require2 (chunk-7D4SUZUM.js?v=71c22fc4:14:50) at electron.js?v=71c22fc4:54:16`
> 
> but while I change the modules import statement into require statement, this error is resolved, I think this api should support import statement `const { webUtils } = require('electron');`

Hi, I can also replicate the same issue

**Comment by Lastor-Chen** [comment]
same issue.

#### Solution:
`webUtils` seems to not support ESM. We can require it in preload and bridge to Browser global window.

```ts
// preload
const { contextBridge, webUtils } = require('electron') as typeof import('electron')

contextBridge.exposeInMainWorld('electronAPI', {
  getPathForFile(file: File) {
    return webUtils.getPathForFile(file)
  },
})

// Browser index.ts
function onInputFileChange(event: Event) {
  const inputElem = event.target as HTMLInputElement
  if (!inputElem?.files.length) return

  const filePath = window.electronAPI.getPathForFile(inputElem.files[0])
  // ...
}
```
